### DISCUSSION TITLE
iPXE Proper Exit for next boot device ?

### DISCUSSION THREAD

**Discussion by a31amit** [ipxe/ipxe#360] [Q&A]
I have recently started using iPXE to boot via DHCP and replacing pxelinux,  I am stuck with a problem with the local disk or the next boot device doesn't properly boot when the default menu timeout. 

I found that iPXE does try the next network card to boot but after that blank screen pop up, where it should be switch to the next bootable device based on defined Boot Order in UEFI. 

I have tried the same thing with - HTTPBOOT, UEFI PXE, UEFI CSM. 

All of them produce the same problems, based on documentation it should be an "exit" or "exit 1" or sanboot for a local drive, nothing gives me with booting with it. 

Any help that how can this be solved? All I needed is a default menu entry that iPXE should exit and UEFI firmware should try the next bootable device according to the order it has been set for.

**Comment by NiKiZe** [comment]
The documentation is what "has been seen to work in the wild" but it is all up to the EFI firmware to do what it wants to do. This is one part of EFI that has no specification.

To get a solution for your specific hardware, you probably need to contact the vendor.

**Reply by a31amit** [reply]
I have tried various hardware from various manufacturers including asus, SuperMicro. I tried today grub2 with Pxe boot method which works fine with exit statement but ipxe doesn’t. I have tried snponly undionly native binaries but all fails for some reasons . This makes me curious that how grub2 achieve this but ipxe can’t ?
